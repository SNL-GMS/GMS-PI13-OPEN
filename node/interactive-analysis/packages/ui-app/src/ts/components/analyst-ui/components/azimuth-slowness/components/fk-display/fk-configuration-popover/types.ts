import { FkTypes } from '@gms/common-graphql';
import { CheckboxSearchListTypes } from '@gms/ui-core-components';
import { FkConfigurationWithUnits, FkUnits } from '~analyst-ui/components/azimuth-slowness/types';

export enum FkConfigurationPopoverPanel {
  DEFAULT,
  ADVANCED
}

export interface FkConfigurationPopoverState {
  openPanel: FkConfigurationPopoverPanel;
  fkUnits: FkUnits;
  normalizeWaveforms: boolean;
  mediumVelocity: number;
  maximumSlowness: number;
  numberOfPoints: number;
  channelCheckboxes: CheckboxSearchListTypes.CheckboxItem[];
  useVerticalChannelOffsets: boolean;
}
export interface FkConfigurationPopoverProps extends FkTypes.FkConfiguration {
  fkUnitDisplayed: FkUnits;
  applyFkConfiguration(configuration: FkConfigurationWithUnits);
  close();
}
