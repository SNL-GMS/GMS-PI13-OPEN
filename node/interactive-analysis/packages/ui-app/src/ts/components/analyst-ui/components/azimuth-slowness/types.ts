import {
  CommonTypes,
  ConfigurationTypes,
  EventTypes,
  FkTypes,
  ProcessingStationTypes,
  SignalDetectionTypes,
  WaveformTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { Client } from '@gms/ui-apollo';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { ChildProps, MutationFunction } from 'react-apollo';
import { FilterType, FkThumbnailSize } from './components/fk-thumbnail-list/fk-thumbnails-controls';
import { LeadLagPairs } from './constants';

export interface FkParams {
  windowParams: FkTypes.WindowParameters;
  frequencyPair: FkTypes.FrequencyBand;
}

export interface LeadLagPairAndString {
  leadLagPairs: LeadLagPairs;
  windowParams: FkTypes.WindowParameters;
}

export enum FkUnits {
  FSTAT = 'FSTAT',
  POWER = 'POWER'
}

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits extends FkTypes.FkConfiguration {
  fkUnitToDisplay: FkUnits;
}

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  client: Client;
  currentStageInterval: AnalystWorkspaceTypes.StageInterval;
  currentTimeInterval: CommonTypes.TimeRange;
  selectedSdIds: string[];
  openEventId: string;
  sdIdsToShowFk: string[];
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSelectedSdIds(ids: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  userInputFkFrequency: FkTypes.FrequencyBand;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  fkInnerContainerWidthPx: number;
  fkFrequencyThumbnails: Immutable.Map<string, FkTypes.FkFrequencyThumbnail[]>;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: MutationFunction<{}>;
  computeFkFrequencyThumbnails: MutationFunction<{}>;
  setWindowLead: MutationFunction<{}>;
  markFksReviewed: MutationFunction<{}>;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  ChildProps<AzimuthSlownessMutations> &
  ProcessingStationTypes.DefaultStationsQueryProps &
  ConfigurationTypes.UIConfigurationQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: ProcessingStationTypes.ProcessingStation[];
  eventsInTimeRange: EventTypes.Event[];
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: EventTypes.Event | undefined;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<string, EventTypes.FeaturePrediction[]>;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  featurePredictionsForDisplayedSignalDetection: EventTypes.FeaturePrediction[];
  distances: EventTypes.LocationToStationDistance[];
  // Redux state as props
  selectedSdIds: string[];
  defaultWaveformFilters: WaveformTypes.WaveformFilter[];
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  userInputFkFrequency: FkTypes.FrequencyBand;
  fkUnitForDisplayedSignalDetection: FkUnits;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  fkInnerContainerWidthPx: number;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  markFksForSdIdsAsReviewed(sdIds: string[]): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(fkInput: FkTypes.FkInput): void;
  changeUserInputFks(
    windowParams: FkTypes.WindowParameters,
    frequencyBand: FkTypes.FrequencyBand
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkUnits): void;

  // Redux functions
  setSelectedSdIds(sdIds: string[]): void;
  setSdIdsToShowFk(sdIds: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}
