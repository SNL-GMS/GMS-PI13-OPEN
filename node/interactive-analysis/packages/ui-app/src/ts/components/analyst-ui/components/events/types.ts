import {
  CommonTypes,
  EventTypes,
  ProcessingStationTypes,
  SignalDetectionTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import { ChildProps, MutationFunction } from 'react-apollo';
import { gmsColors, semanticColors } from '~scss-config/color-preferences';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}
export const eventFilterToColorMap: Map<any, string> = new Map();
eventFilterToColorMap.set(EventFilters.EDGE, gmsColors.gmsBackground);
eventFilterToColorMap.set(EventFilters.COMPLETED, semanticColors.analystComplete);

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction<{}>;
  saveEvent: MutationFunction<{}>;
  saveAllModifiedEvents: MutationFunction<{}>;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}
export interface SignalDetectionHypothesisWithStation
  extends SignalDetectionTypes.SignalDetectionHypothesis {
  stationName?: string;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  EventTypes.EventsInTimeRangeQueryProps &
  ProcessingStationTypes.DefaultStationsQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  CommonTypes.WorkspaceStateProps;
