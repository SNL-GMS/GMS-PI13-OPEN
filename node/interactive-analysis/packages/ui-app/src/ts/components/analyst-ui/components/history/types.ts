import { CacheTypes, CommonTypes, EventTypes } from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import React from 'react';
import { ChildProps, MutationFunction } from 'react-apollo';
import { HistoryEntryAction } from './history-stack/types';

/**
 * Mutations used by the History component
 */
export interface HistoryMutations {
  undoHistory: MutationFunction<CacheTypes.UndoHistoryMutationResult>;
  redoHistory: MutationFunction<CacheTypes.RedoHistoryMutationResult>;
  undoHistoryById: MutationFunction<CacheTypes.UndoHistoryByIdMutationResult>;
  redoHistoryById: MutationFunction<CacheTypes.RedoHistoryByIdMutationResult>;
  undoEventHistory: MutationFunction<CacheTypes.UndoEventHistoryMutationResult>;
  redoEventHistory: MutationFunction<CacheTypes.RedoEventHistoryMutationResult>;
  undoEventHistoryById: MutationFunction<CacheTypes.UndoEventHistoryByIdMutationResult>;
  redoEventHistoryById: MutationFunction<CacheTypes.RedoEventHistoryByIdMutationResult>;
}

/**
 * History Redux Props
 */
export interface HistoryReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  historyActionInProgress: number;
  openEventId: string;
  currentTimeInterval: CommonTypes.TimeRange;
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  keyPressActionQueue: AnalystWorkspaceTypes.KeyAction[];
  setKeyPressActionQueue(actions: AnalystWorkspaceTypes.KeyAction[]): void;
  incrementHistoryActionInProgress();
  decrementHistoryActionInProgress();
}

/**
 * History Props
 */
export type HistoryProps = HistoryReduxProps &
  ChildProps<HistoryMutations> &
  CacheTypes.HistoryQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State for the HistoryComponent
 */
export interface HistoryState {
  actionIntentPointer: HistoryEntryPointer;
}
/**
 * History Panel Props
 * optional non-ideal-state element to render in the children history-stacks
 */
export interface HistoryPanelProps {
  nonIdealState?: NonIdealHistoryState;
}

/**
 * Keeps track of which HistoryEntryAction is being target for an undo/redo action
 * and provides a check function isIncluded for checking if other actions would be
 * included based on whether the target is an event or global undo/redo.
 */
export interface HistoryEntryPointer {
  entryId: string;
  entryType: HistoryEntryAction;
  isEventMode: boolean;
  isIncluded?(entry: CacheTypes.History): boolean;
  isChangeIncluded(change: CacheTypes.HistoryChange): boolean;
}

/**
 * The type for the data used by history context
 */
export interface HistoryContextData {
  glContainer: GoldenLayout.Container;
  historyList: CacheTypes.History[];
  historyActionIntent: HistoryEntryPointer;
  openEventId: string;
  eventsInTimeRange: EventTypes.Event[];
  historyActionInProgress: number;
  redoEventHistory(quantity?: number): void;
  redoEventHistoryById(id: string): void;
  redoHistory(quantity?: number): void;
  redoHistoryById(id: string): void;
  setHistoryActionIntent(pointer: HistoryEntryPointer): void;
  undoEventHistory(quantity?: number): void;
  undoEventHistoryById(id: string): void;
  undoHistory(quantity?: number): void;
  undoHistoryById(id: string): void;
}

/**
 * History Context to provide access to these high-level props to the History-Component tree.
 * Here, we instantiate the Context variable and set up the defaults for the history context
 */
export const HistoryContext: React.Context<HistoryContextData> = React.createContext<
  HistoryContextData
>({
  glContainer: undefined,
  openEventId: undefined,
  eventsInTimeRange: [],
  undoEventHistory: undefined,
  undoEventHistoryById: undefined,
  redoEventHistory: undefined,
  redoEventHistoryById: undefined,
  undoHistory: undefined,
  undoHistoryById: undefined,
  redoHistory: undefined,
  redoHistoryById: undefined,
  historyActionIntent: undefined,
  setHistoryActionIntent: undefined,
  historyActionInProgress: 0,
  historyList: []
});

export interface NonIdealHistoryState {
  loadingEvent: JSX.Element;
  loadingHistory: JSX.Element;
}
