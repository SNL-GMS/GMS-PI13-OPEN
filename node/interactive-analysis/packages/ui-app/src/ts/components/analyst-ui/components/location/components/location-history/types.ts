import { EventTypes } from '@gms/common-graphql';
import { Row } from '@gms/ui-core-components';
import { AnalystWorkspaceTypes } from '@gms/ui-state';

/**
 * LocationHistory Props
 */
export interface LocationHistoryProps {
  // Current open event
  event: EventTypes.Event;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  setSelectedLocationSolution(locationSolutionSetId: string, locationSolutionId: string): void;
  setSelectedPreferredLocationSolution(
    preferredLocationSolutionSetId: string,
    preferredLocationSolutionId: string
  ): void;
}
/**
 * Location History State
 */
// tslint:disable-next-line: no-empty-interface
export interface LocationHistoryState {}

export interface LocationHistoryRow extends Row {
  locationSolutionId: string;
  locationSetId: string;
  locType: string;
  latestLSSId: string;
  lat: number;
  lon: number;
  depth: number;
  time: string;
  restraint: string;
  smajax: number;
  sminax: number;
  strike: number;
  stdev: number;
  depthRestraintType: EventTypes.DepthRestraintType;
  count: number;
  selectedLocationSolutionSetId: string;
  isLocationSolutionSetPreferred?: boolean;
  isLastInLSSet?: boolean;
  isFirstInLSSet?: boolean;
  preferred?: boolean;
  locationGroup?: LocationHistoryRow[];
  setSelectedPreferredLocationSolution(
    preferredLocationSolutionSetId: string,
    preferredLocationSolutionId: string
  ): void;
  setToSave(locationSolutionSetId: string, locationSolutionId: string): void;
}
