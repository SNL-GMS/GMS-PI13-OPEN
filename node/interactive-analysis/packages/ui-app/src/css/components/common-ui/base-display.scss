.base-display {
  display: flex;
  position: relative;
  flex-direction: column;
  &.full-width-height {
    height: 100%;
    width: 100%;
    flex-grow: 1;
  }
}
