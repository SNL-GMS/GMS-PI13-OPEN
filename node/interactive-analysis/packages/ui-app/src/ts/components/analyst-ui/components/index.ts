export { ReduxApolloWaveformDisplay as WaveformDisplay } from './waveform-display';
export { Events } from './events';
export { History } from './history';
export { SignalDetections } from './signal-detections';
export { Workflow } from './workflow';
export { ReduxApolloMap as Map } from './map';
export { Location } from './location';
export { Magnitude } from './magnitude';

export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness/azimuth-slowness-container';
