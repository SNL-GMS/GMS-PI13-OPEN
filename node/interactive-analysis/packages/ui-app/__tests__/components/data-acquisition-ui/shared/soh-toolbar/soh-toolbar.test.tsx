import { SohStatusSummary } from '@gms/common-graphql/lib/graphql/soh/types';
import React from 'react';
import { BaseDisplayContext } from '../../../../../src/ts/components/common-ui/components/base-display';
import {
  SohToolbar,
  SohToolbarProps
} from '../../../../../src/ts/components/data-acquisition-ui/shared/toolbars/soh-toolbar';
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('Soh toolbar component', () => {
  const statusesToDisplay: Map<any, boolean> = new Map().set(SohStatusSummary.GOOD, true);
  const item: any = {
    dropdownOptions: {},
    value: {},
    custom: false,
    onChange: jest.fn()
  };
  const sohToolbarProps: SohToolbarProps = {
    statusesToDisplay,
    updateIntervalSecs: 1,
    updatedAt: 10000,
    widthPx: 1000,
    leftItems: [item],
    rightItems: [],
    statusFilterText: '',
    setStatusesToDisplay: jest.fn(),
    toggleHighlight: jest.fn(),
    sohStationStaleTimeMS: 30000,
    displayTimeWarning: false,
    statusFilterTooltip: 'test tooltip'
  };
  const sohToolbar = Enzyme.mount(
    <BaseDisplayContext.Provider
      value={{
        glContainer: { width: 150, height: 150 } as any,
        widthPx: 150,
        heightPx: 150
      }}
    >
      <SohToolbar {...sohToolbarProps} />
    </BaseDisplayContext.Provider>
  );

  it('should be defined', () => {
    expect(SohToolbar).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(sohToolbar).toMatchSnapshot();
  });
});
