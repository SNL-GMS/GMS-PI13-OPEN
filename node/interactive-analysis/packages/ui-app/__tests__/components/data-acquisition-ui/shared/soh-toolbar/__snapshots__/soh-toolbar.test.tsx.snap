// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soh toolbar component should match snapshot 1`] = `
<Component
  displayTimeWarning={false}
  leftItems={
    Array [
      Object {
        "custom": false,
        "dropdownOptions": Object {},
        "onChange": [MockFunction],
        "rank": 2,
        "value": Object {},
      },
    ]
  }
  rightItems={Array []}
  setStatusesToDisplay={[MockFunction]}
  sohStationStaleTimeMS={30000}
  statusFilterText=""
  statusFilterTooltip="test tooltip"
  statusesToDisplay={
    Map {
      "GOOD" => true,
    }
  }
  toggleHighlight={[MockFunction]}
  updateIntervalSecs={1}
  updatedAt={10000}
  widthPx={1000}
>
  <Component
    items={
      Array [
        Object {
          "hasIssue": false,
          "label": "Last Updated",
          "rank": 1,
          "style": Object {
            "marginLeft": "1em",
          },
          "tooltip": "Most recent SOH data received",
          "tooltipForIssue": "SOH data has not updated in over 30 seconds",
          "type": "LabelValue",
          "value": "1970/01/01 02:46:40",
          "widthPx": 400,
        },
        Object {
          "label": "Update Interval",
          "rank": 2,
          "tooltip": "Interval at which SOH data is processed",
          "type": "LabelValue",
          "value": "1 second",
          "widthPx": 400,
        },
      ]
    }
    itemsLeft={
      Array [
        Object {
          "colors": Map {
            "GOOD" => "#fff",
            "MARGINAL" => "#fff",
            "BAD" => "#fff",
            "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
          },
          "cyData": "filter-soh",
          "enumKeysToDisplayStrings": Map {
            "GOOD" => "Good",
            "MARGINAL" => "Marginal",
            "BAD" => "Bad",
            "NONE" => "None",
          },
          "enumOfKeys": Object {
            "BAD": "BAD",
            "GOOD": "GOOD",
            "MARGINAL": "MARGINAL",
            "NONE": "NONE",
          },
          "label": "",
          "menuLabel": "",
          "onChange": [MockFunction],
          "onPopUp": [Function],
          "onPopoverDismissed": [Function],
          "rank": 1,
          "tooltip": "test tooltip",
          "type": "CheckboxList",
          "values": Map {
            "GOOD" => true,
          },
          "widthPx": 200,
        },
        Object {
          "custom": false,
          "dropdownOptions": Object {},
          "onChange": [MockFunction],
          "rank": 2,
          "value": Object {},
        },
      ]
    }
    widthPx={1000}
  >
    <ToolbarComponent
      items={
        Array [
          Object {
            "hasIssue": false,
            "label": "Last Updated",
            "rank": 1,
            "style": Object {
              "marginLeft": "1em",
            },
            "tooltip": "Most recent SOH data received",
            "tooltipForIssue": "SOH data has not updated in over 30 seconds",
            "type": "LabelValue",
            "value": "1970/01/01 02:46:40",
            "widthPx": 400,
          },
          Object {
            "label": "Update Interval",
            "rank": 2,
            "tooltip": "Interval at which SOH data is processed",
            "type": "LabelValue",
            "value": "1 second",
            "widthPx": 400,
          },
        ]
      }
      itemsLeft={
        Array [
          Object {
            "colors": Map {
              "GOOD" => "#fff",
              "MARGINAL" => "#fff",
              "BAD" => "#fff",
              "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
            },
            "cyData": "filter-soh",
            "enumKeysToDisplayStrings": Map {
              "GOOD" => "Good",
              "MARGINAL" => "Marginal",
              "BAD" => "Bad",
              "NONE" => "None",
            },
            "enumOfKeys": Object {
              "BAD": "BAD",
              "GOOD": "GOOD",
              "MARGINAL": "MARGINAL",
              "NONE": "NONE",
            },
            "label": "",
            "menuLabel": "",
            "onChange": [MockFunction],
            "onPopUp": [Function],
            "onPopoverDismissed": [Function],
            "rank": 1,
            "tooltip": "test tooltip",
            "type": "CheckboxList",
            "values": Map {
              "GOOD" => true,
            },
            "widthPx": 200,
          },
          Object {
            "custom": false,
            "dropdownOptions": Object {},
            "onChange": [MockFunction],
            "rank": 2,
            "value": Object {},
          },
        ]
      }
      minWhiteSpacePx={1}
      toolbarWidthPx={960}
      widthPx={1000}
    >
      <div
        className="toolbar"
        style={
          Object {
            "width": "960px",
          }
        }
      >
        <div
          className="toolbar__left-group"
        >
          <div
            className="toolbar-item toolbar-item__left"
            key="1"
          >
            <Component
              addToPopoverMap={[Function]}
              hasIssue={false}
              item={
                Object {
                  "colors": Map {
                    "GOOD" => "#fff",
                    "MARGINAL" => "#fff",
                    "BAD" => "#fff",
                    "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
                  },
                  "cyData": "filter-soh",
                  "enumKeysToDisplayStrings": Map {
                    "GOOD" => "Good",
                    "MARGINAL" => "Marginal",
                    "BAD" => "Bad",
                    "NONE" => "None",
                  },
                  "enumOfKeys": Object {
                    "BAD": "BAD",
                    "GOOD": "GOOD",
                    "MARGINAL": "MARGINAL",
                    "NONE": "NONE",
                  },
                  "label": "",
                  "menuLabel": "",
                  "onChange": [MockFunction],
                  "onPopUp": [Function],
                  "onPopoverDismissed": [Function],
                  "rank": 1,
                  "tooltip": "test tooltip",
                  "type": "CheckboxList",
                  "values": Map {
                    "GOOD" => true,
                  },
                  "widthPx": 200,
                }
              }
              key="1"
            >
              <div
                style={Object {}}
              >
                <PopoverButtonComponent
                  cyData="filter-soh"
                  key="1"
                  label=""
                  onClick={[Function]}
                  onPopoverDismissed={[Function]}
                  popupContent={
                    <CheckboxList
                      checkboxEnum={
                        Object {
                          "BAD": "BAD",
                          "GOOD": "GOOD",
                          "MARGINAL": "MARGINAL",
                          "NONE": "NONE",
                        }
                      }
                      enumKeysToDisplayStrings={
                        Map {
                          "GOOD" => "Good",
                          "MARGINAL" => "Marginal",
                          "BAD" => "Bad",
                          "NONE" => "None",
                        }
                      }
                      enumToCheckedMap={
                        Map {
                          "GOOD" => true,
                        }
                      }
                      enumToColorMap={
                        Map {
                          "GOOD" => "#fff",
                          "MARGINAL" => "#fff",
                          "BAD" => "#fff",
                          "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
                        }
                      }
                      onChange={[Function]}
                    />
                  }
                  tooltip="test tooltip"
                  widthPx={200}
                >
                  <div
                    data-cy="filter-soh"
                  >
                    <Blueprint3.Button
                      active={false}
                      alignText="left"
                      className="toolbar-button"
                      onClick={[Function]}
                      style={
                        Object {
                          "width": "200px",
                        }
                      }
                      title="test tooltip"
                    >
                      <button
                        className="bp3-button bp3-align-left toolbar-button"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        style={
                          Object {
                            "width": "200px",
                          }
                        }
                        title="test tooltip"
                        type="button"
                      >
                        <Blueprint3.Icon
                          key="leftIcon"
                        />
                        <span
                          className="bp3-button-text"
                          key="text"
                        >
                          <span />
                          <Blueprint3.Icon
                            icon="chevron-down"
                            title={false}
                          >
                            <span
                              className="bp3-icon bp3-icon-chevron-down"
                              icon="chevron-down"
                            >
                              <svg
                                data-icon="chevron-down"
                                height={16}
                                viewBox="0 0 16 16"
                                width={16}
                              >
                                <path
                                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                  fillRule="evenodd"
                                  key="0"
                                />
                              </svg>
                            </span>
                          </Blueprint3.Icon>
                        </span>
                        <Blueprint3.Icon
                          key="rightIcon"
                        />
                      </button>
                    </Blueprint3.Button>
                  </div>
                </PopoverButtonComponent>
              </div>
            </Component>
          </div>
          <div
            className="toolbar-item toolbar-item__left"
            key="2"
          >
            <Component
              addToPopoverMap={[Function]}
              hasIssue={false}
              item={
                Object {
                  "custom": false,
                  "dropdownOptions": Object {},
                  "onChange": [MockFunction],
                  "rank": 2,
                  "value": Object {},
                }
              }
              key="2"
            >
              <div />
            </Component>
          </div>
        </div>
        <div
          className="toolbar__center-group"
        >
          <div
            className="toolbar__whitespace"
            style={
              Object {
                "width": "944px",
              }
            }
          />
        </div>
        <div
          className="toolbar__right-group"
        >
          <div
            className="toolbar-item
                    "
            key="1"
          >
            <Component
              addToPopoverMap={[Function]}
              hasIssue={false}
              item={
                Object {
                  "hasIssue": false,
                  "label": "Last Updated",
                  "rank": 1,
                  "style": Object {
                    "marginLeft": "1em",
                  },
                  "tooltip": "Most recent SOH data received",
                  "tooltipForIssue": "SOH data has not updated in over 30 seconds",
                  "type": "LabelValue",
                  "value": "1970/01/01 02:46:40",
                  "widthPx": 400,
                }
              }
              key="1"
            >
              <div
                className="toolbar-label-value"
                style={
                  Object {
                    "marginLeft": "1em",
                  }
                }
              >
                <Component
                  label="Last Updated"
                  tooltip="Most recent SOH data received"
                  value="1970/01/01 02:46:40"
                >
                  <div
                    className="label-value-container"
                  >
                    <div
                      className="label-value__label"
                    >
                      Last Updated: 
                    </div>
                    <div
                      className="label-value__value"
                      style={
                        Object {
                          "color": "",
                        }
                      }
                      title="Most recent SOH data received"
                    >
                      1970/01/01 02:46:40
                    </div>
                  </div>
                </Component>
              </div>
            </Component>
          </div>
          <div
            className="toolbar-item
                    "
            key="2"
          >
            <Component
              addToPopoverMap={[Function]}
              hasIssue={false}
              item={
                Object {
                  "label": "Update Interval",
                  "rank": 2,
                  "tooltip": "Interval at which SOH data is processed",
                  "type": "LabelValue",
                  "value": "1 second",
                  "widthPx": 400,
                }
              }
              key="2"
            >
              <div
                className="toolbar-label-value"
                style={Object {}}
              >
                <Component
                  label="Update Interval"
                  tooltip="Interval at which SOH data is processed"
                  value="1 second"
                >
                  <div
                    className="label-value-container"
                  >
                    <div
                      className="label-value__label"
                    >
                      Update Interval: 
                    </div>
                    <div
                      className="label-value__value"
                      style={
                        Object {
                          "color": "",
                        }
                      }
                      title="Interval at which SOH data is processed"
                    >
                      1 second
                    </div>
                  </div>
                </Component>
              </div>
            </Component>
          </div>
        </div>
      </div>
    </ToolbarComponent>
  </Component>
</Component>
`;
